### PR TITLE
Bump to v1.10.0rc1

### PR DESCRIPTION
**Pull request by tomvothecoder**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

- Related to https://github.com/E3SM-Project/e3sm_to_cmip/pull/194
- Constrain python >=3.9

### CODE REVIEW

**Comment by tomvothecoder** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+{% set version = "1.10.0rc1" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://github.com/E3SM-Project/{{ name }}/archive/v{{ version }}.tar.gz
-  sha256: f37a498948c35a322ebdf67059d3918664b60f61f013fbc9e4dfb92e41467a43
+  sha256:
```
Comment (tomvothecoder) [inline recipe/meta.yaml:10]: TODO: Add sha256 of GH release

**Comment by tomvothecoder** [comment]
Hey @xylar, you said to reach out if we're making stable releases and release candidates. Is there some special instructions for each type (e.g., labels to use for conda-forge)?

**Comment by xylar** [comment]
@tomvothecoder, we want to change e3sm_dev to e3sm_to_cimp_dev here:
https://github.com/conda-forge/e3sm_to_cmip-feedstock/blob/dev/recipe/conda_build_config.yaml#L2

I can push a commit to do that.

**Comment by xylar** [comment]
@tomvothecoder, this needs to be a PR to the `dev` branch.  Can you rebase onto that branch.  Then, I can edit the file (or you can) that I pointed to above.

**Comment by tomvothecoder** [comment]
> @tomvothecoder, this needs to be a PR to the `dev` branch. Can you rebase onto that branch. Then, I can edit the file (or you can) that I pointed to above.

Ahh that's right. It's been awhile since I made an `rc` and I forgot to do this.

I'll handle both of your comments and tag you when it's done.

**Comment by tomvothecoder** [comment]
@conda-forge-admin, please rerender

**Comment by tomvothecoder** [comment]
@conda-forge-admin, please rerender

**Comment by tomvothecoder** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
@tomvothecoder, looks good.  Just waiting on the `sha256` value, I guess.

**Comment by tomvothecoder** [comment]
> @tomvothecoder, looks good. Just waiting on the `sha256` value, I guess.

I literally just pushed the shasum at the same time as your comment lol.

**Review by xylar — APPROVED** [review]

**Pull request merged**